src/appleseed/renderer/modeling/camera/projectpoints.cpp

```diff
@@ -0,0 +1,142 @@
+
+//
+// This source file is part of appleseed.
+// Visit https://appleseedhq.net/ for additional information and resources.
+//
+// This software is released under the MIT license.
+//
+// Copyright (c) 2018 Esteban Tovagliari, The appleseedhq Organization
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+// Interface header.
+#include "projectpoints.h"
+
+// appleseed.renderer headers.
+#include "renderer/modeling/entity/onframebeginrecorder.h"
+#include "renderer/modeling/entity/onrenderbeginrecorder.h"
+#include "renderer/modeling/frame/frame.h"
+#include "renderer/modeling/project/project.h"
+#include "renderer/modeling/scene/scene.h"
+#include "renderer/utility/paramarray.h"
+
+// appleseed.foundation headers.
+#include "foundation/utility/iostreamop.h"
+
+// Standard headers.
+#include <string>
+
+using namespace foundation;
+using namespace std;
+
+namespace renderer
+{
+
+struct ProjectPoints::Impl
+{
+    Impl()
+      : m_camera(nullptr)
+    {
+    }
+
+    Camera*                     m_camera;
+    auto_release_ptr<Project>   m_project;
+    OnRenderBeginRecorder       m_render_begin_recorder;
+    OnFrameBeginRecorder        m_frame_begin_recorder;
+};
+
+ProjectPoints::ProjectPoints(
+    auto_release_ptr<Camera> camera,
+    const Vector2u&          resolution)
+  : impl(new Impl())
+{
+    auto_release_ptr<Scene> scene = SceneFactory::create();
+
+    const string camera_name = camera->get_name();
```

Comment:
Get rid of `camera_name` variable, use `camera->get_name()` directly when creating the frame.

Reply:
camera is an auto_release_ptr and it is gone after I insert it into the scene.
I need to save the name before inserting it into the scene.

Reply:
Ok.